### PR TITLE
Resolve #170 again using gcr.io/distroless/static 

### LINKED ISSUE #170: Docker image cannot `convert`
When running `convert` using the Docker image, it fails with an error like
```
2024/06/08 20:18:17 main.go:177: Failed to create temp file, open /tmp/pmtiles3706954202: no such file or directory
```

This is probably because there is no `/tmp` directory in the Docker image.

To reproduce:

Assuming there is an `output.mbtiles` file in the current directory:
```sh
docker run --rm -v .:/data protomaps/go-pmtiles convert /data/output.mbtiles /data/output.pmtiles
```

The following can be used to produce such an `output.mbtiles`:
```sh
wget https://raw.githubusercontent.com/onthegomap/planetiler/main/planetiler-custommap/src/main/resources/samples/power.yml
docker run -v .:/data ghcr.io/onthegomap/planetiler:latest generate-custom --download --schema=/data/power.yml
```

### PR DESCRIPTION
**Pull request by zstadler**
Following @bdon's https://github.com/protomaps/go-pmtiles/issues/170#issuecomment-2162843755

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,9 +3,8 @@ COPY . /workspace
 WORKDIR /workspace
 ENV CGO_ENABLED=0
 RUN go build -o /workspace/go-pmtiles
-FROM scratch
+FROM gcr.io/distroless/static
 COPY --from=builder /workspace/go-pmtiles /go-pmtiles
 COPY --from=builder etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```
Comment (bdon) [inline Dockerfile:8]: Is it possible we don't need this anymore with `distroless` too?

Comment (zstadler) [reply Dockerfile:8]: You're probably right, but I don't know how does this line affects the program. I also don't know how to test that nothing is broken after removing it from `Dockerfile`.

I've commented-out this line in a `scratch`-based `Dockerfile`:
```
FROM golang:1.21.8-alpine3.19 AS builder
COPY . /workspace
WORKDIR /workspace
ENV CGO_ENABLED=0
RUN go build -o /workspace/go-pmtiles
FROM scratch
COPY --from=builder /workspace/go-pmtiles /go-pmtiles
# COPY --from=builder etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
ADD root_tmp.tgz /
EXPOSE 8080
ENTRYPOINT ["/go-pmtiles"]
```

and that did not seem to make any difference for `go-pmtiles convert` and `go-pmtiles serve`.

Comment (zstadler) [reply Dockerfile]: In any case, there are certificates in the base image, so no need to add them.

Comment (bdon) [reply Dockerfile]: The certs are needed for making `pmtiles serve`, `pmtiles show` connections to HTTPS endpoints

**Comment by bdon** [comment]
Thanks!

**Comment by bdon** [comment]
I tagged 1.20.1: https://github.com/protomaps/go-pmtiles/pkgs/container/go-pmtiles

**Pull request merged**